prov/sockets/src/sock_progress.c

```diff
@@ -1189,9 +1193,20 @@ static int sock_pe_progress_buffered_rx(struct sock_rx_ctx *rx_ctx)
 	    dlist_empty(&rx_ctx->rx_buffered_list))
 		return 0;
 
-	for (entry = rx_ctx->rx_buffered_list.next;
-	     entry != &rx_ctx->rx_buffered_list;) {
-
+        if (depth < 0) {
+            /* check entire buffered list */
+            entry = &rx_ctx->rx_buffered_list;
+        } else {
+            /* continue where last time left off */
+            entry = rx_ctx->progress_start;
+        }
+        if (entry == &rx_ctx->rx_buffered_list) {
+            entry = entry->next;
+        }
+        for (int i_prog = 0; depth < 0 || i_prog < SOCK_EP_MAX_PROGRESS_CNT; i_prog++) {
```

Comment:
Just call the iteration variable 'i'.  Please declare all variables at the top of the function.

Reply:
This should be
```
for (i = 0; i < depth && entry != &rx_ctx->rx_buffered_list ; i++) {
```
depth should be set to either a system max or max progress cnt.  Additionally, depth should be an unsigned value, such as size_t.